pkg/apis/batch/validation/validation.go

```diff
@@ -384,9 +386,43 @@ func validateTimeZone(timeZone *string, fldPath *field.Path) field.ErrorList {
 		allErrs = append(allErrs, field.Invalid(fldPath, timeZone, err.Error()))
 	}
 
+	if err := validateTimeZoneCase(*timeZone); err != nil {
+		allErrs = append(allErrs, field.Invalid(fldPath, timeZone, err.Error()))
+	}
+
 	return allErrs
 }
 
+// validateTimeZoneCase compares the timeZone with golang's builtin database
+// to ensure the caseing is correctly recognized. This only applies to MacOS
+// systems with case insensitive filesystem, where timezone can be
+// inproperly recognized.
+// See https://github.com/golang/go/issues/21512 for more details.
+func validateTimeZoneCase(timeZone string) error {
+	if runtime.GOOS == "darwin" {
```

Comment:
wouldn't it be better to compare the lowercase strings and have a case for darwin in the for loop?

Reply:
not needed the values are guaranteed by the language https://pkg.go.dev/runtime#pkg-constants

Reply:
I meant comparing the timezones by lowercase for darwin

Reply:
Yeah, but our tests explicitly verify the case correctness, as you see in `validation_tests.go` so the whole point is to make sure that the case exactly matches what we have in the database file. If you check https://go.dev/src/time/zoneinfo_unix.go, specifically this fragment:
```golang
var zoneSources = []string{
	"/usr/share/zoneinfo/",
	"/usr/share/lib/zoneinfo/",
	"/usr/lib/locale/TZ/",
	runtime.GOROOT() + "/lib/time/zoneinfo.zip",
}
```
you'll notice that the mechanism first checks OS-provided time zone database, which is just a list of directories with files, where the name matches exactly the name of the time zone. The problem is that for MacOS with case insensitive file system `America/NewYork` is equal to `AMERICA/Newyork` which causes a problem for our validation tests. Thus to make sure the case is exactly we fallback to verifying the golang provided database which is a zip file, thus we can rely on casing for that. I hope that makes more sense now :nerd_face:

Reply:
Thanks for explanation :)